proxy/http2/Http2Stream.cc

```diff
@@ -356,8 +356,6 @@ Http2Stream::transaction_done()
 
   if (closed) {
     // Safe to initiate SSN_CLOSE if this is the last stream
-    if (cross_thread_event)
-      cross_thread_event->cancel();
     // Schedule the destroy to occur after we unwind here.  IF we call directly, may delete with reference on the stack.
```

Comment:
If do_io_close set the event again, would it make sense to check if the even is cancelled rather than removing this code? In our tests, this doesn't happen, but I'm not familiar with the code to be sure it never happens.